src/common/transformations/src/transformations/convert_precision.cpp

```diff
@@ -527,85 +604,121 @@ bool fuse_type_to_multiclass_nms(const std::shared_ptr<ngraph::Node>& node, ngra
         return false;
     }
 
-    if ((idx == 1 || idx == 2) && (to == ov::element::i32 || to == ov::element::i64)) {
-        nms->set_output_type(to);
-        return true;
+    auto it = precisions.find(node->get_output_element_type(1));
+    if (it != precisions.end()) {
+        const auto& to = it->second;
+        if (to == ov::element::i32 || to == ov::element::i64) {
+            nms->set_output_type(to);
+            return true;
+        }
     }
 
     return false;
 }
 
-bool fuse_type_to_generate_proposals(const std::shared_ptr<ngraph::Node>& node, ngraph::element::Type to, size_t idx) {
+bool fuse_type_to_generate_proposals(const std::shared_ptr<ngraph::Node>& node, const precisions_map& precisions) {
     auto generate_proposals = ov::as_type_ptr<opset9::GenerateProposals>(node);
     if (!generate_proposals) {
         return false;
     }
 
-    if ((idx == 2) && (to == ov::element::i32 || to == ov::element::i64)) {
-        generate_proposals->set_roi_num_type(to);
-        return true;
+    auto it = precisions.find(node->get_output_element_type(2));
+    if (it != precisions.end()) {
```

Comment:
I believe we can implement a separate helper function for all these cases,e.g.
```
bool update_type(size_t idx, function update_method) {
auto it = precisions.find(node->get_output_element_type(idx));
    if (it != precisions.end()) {
        const auto& to = it->second;
        if (to == ov::element::i32 || to == ov::element::i64) {
            update_method();
            return true;
        }
    }
    }
```
    
and after that we will be able to simplify all these calls, e.g.
`update_type(2, [](){generate_proposals->set_roi_num_type(to);}) 
`
what do you think?

Reply:
done